tests/test_math/test_compactFock.py

```diff
@@ -38,54 +43,63 @@ def random_ABC(draw, M):
 @given(random_ABC(M=3))
 def test_compactFock_diagonal(A_B_G0):
     """Test getting Fock amplitudes if all modes are detected (math.hermite_renormalized_diagonal)"""
-    for cutoffs in allowed_cutoffs((7, 7, 7)):
-        A, B, G0 = A_B_G0  # Create random state (M mode Gaussian state with displacement)
-
-        # Vanilla MM
-        G_ref = math.hermite_renormalized(
-            math.conj(-A), math.conj(B), math.conj(G0), shape=list(cutoffs) * 2
-        ).numpy()  # note: shape=[C1,C2,C3,...,C1,C2,C3,...]
-
-        # Extract diagonal amplitudes from vanilla MM
-        ref_diag = np.zeros(cutoffs, dtype=np.complex128)
-        for inds in np.ndindex(*cutoffs):
-            inds_expanded = list(inds) + list(inds)  # a,b,c,a,b,c
-            ref_diag[inds] = G_ref[tuple(inds_expanded)]
-
-        # New MM
-        G_diag = math.hermite_renormalized_diagonal(
-            math.conj(-A), math.conj(B), math.conj(G0), cutoffs
-        )
-        assert np.allclose(ref_diag, G_diag)
-
+    for precision in allowed_values_precision:
```

Comment:
How do I use` @hypothesis.given` and `@pytest.mark.parametrize` at the same time?

Reply:
I found some solutions on stackoverflow, but I think they are less intuitive than what I currently have.